src/MongoDB/Session.c

```diff
@@ -256,12 +256,100 @@ static PHP_METHOD(Session, getServer)
 
 	/* For sessions without a pinned server, 0 is returned. */
 	if (!server_id) {
-	    RETURN_NULL();
+		RETURN_NULL();
 	}
 
 	phongo_server_init(return_value, mongoc_client_session_get_client(intern->client_session), server_id TSRMLS_CC);
 } /* }}} */
 
+/* {{{ proto array|null MongoDB\Driver\Session::getTransactionOptions()
+   Returns options for the currently running transaction */
+static PHP_METHOD(Session, getTransactionOptions)
+{
+	php_phongo_session_t*     intern;
+	mongoc_transaction_opt_t* opts;
+
+	intern = Z_SESSION_OBJ_P(getThis());
+	SESSION_CHECK_LIVELINESS(intern, "getTransactionOptions")
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	opts = mongoc_session_opts_get_transaction_opts(intern->client_session);
+
+	if (!opts) {
+		return;
+	}
+
+	array_init_size(return_value, 4);
+
+	ADD_ASSOC_LONG_EX(return_value, "maxCommitTimeMS", mongoc_transaction_opts_get_max_commit_time_ms(opts));
+
+	{
+		const mongoc_read_concern_t* read_concern = mongoc_transaction_opts_get_read_concern(opts);
+
+		if (read_concern) {
+#if PHP_VERSION_ID >= 70000
+			zval zread_concern;
+
+			phongo_readconcern_init(&zread_concern, read_concern TSRMLS_CC);
+			ADD_ASSOC_ZVAL_EX(return_value, "readConcern", &zread_concern);
+#else
+			zval* zread_concern = NULL;
+			MAKE_STD_ZVAL(zread_concern);
+
+			phongo_readconcern_init(zread_concern, read_concern TSRMLS_CC);
+			ADD_ASSOC_ZVAL_EX(return_value, "readConcern", zread_concern);
+#endif
+		} else {
+			ADD_ASSOC_NULL_EX(return_value, "readConcern");
```

Comment:
Using these as argument to `startTransaction()` won't be possible as the helper returns `null` when no transaction is in progress:

> A newly allocated `mongoc_transaction_opt_t` that should be freed with `mongoc_transaction_opts_destroy` or NULL if the session is not in a transaction.

([Source](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/doc/mongoc_session_opts_get_transaction_opts.rst))

We don't expose the default transaction options in the session (obtainable using `mongoc_session_opts_get_default_transaction_opts`), so I don't see people using the returned structure for this purpose.

On the case of omitting them when not set, I'm a bit conflicted. I looked at the `*_get_debug_info()` functions for guidance and decided that it would be sensible to always have the same structure for the return value. On the other hand, this does not mean much for downstream libraries as they can't rely on array keys being present anyways, so they would need to add checks when accessing the structure, regardless of whether we include keys for empty values or not.

---

An alternative to this could be to not return an array, but instead create an object containing these values:
```
class TransactionOptions
{
    public function getMaxCommitTimeMS(): ?int;

    public function getReadConcern(): ?ReadConcern;

    public function getReadPreference(): ?ReadPreference;

    public function getWriteConcern(): ?WriteConcern;
}
```

This would give users a fixed API to work with, while keeping it flexible enough to adhere to the [specification for `TransactionOptions`](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#transactionoptions):
> `TransactionOptions` MUST be designed such that future options can be added without breaking backward compatibility.

A "downside" to this approach would be the downstream changes, as users would expect this `TransactionOptions` object to be accepted in `Client::startSession` (for `defaultTransactionOptions`) and `Session::startTransaction`. The advantage of such an object is that users that want to use strict typing will have an easier time checking their code as they are no longer relying on arrays containing options. What do you think?